sphinx/application.py

```diff
@@ -664,34 +1023,83 @@ def add_lexer(self, alias, lexer):
 
     def add_autodocumenter(self, cls):
         # type: (Any) -> None
+        """Register a new documenter class for the autodoc extension.
+
+        Add *cls* as a new documenter class for the :mod:`sphinx.ext.autodoc`
+        extension.  It must be a subclass of
+        :class:`sphinx.ext.autodoc.Documenter`.  This allows to auto-document
+        new types of objects.  See the source of the autodoc module for
+        examples on how to subclass :class:`Documenter`.
+
+        .. todo:: Add real docs for Documenter and subclassing
+
+        .. versionadded:: 0.6
+        """
         logger.debug('[app] adding autodocumenter: %r', cls)
         from sphinx.ext.autodoc.directive import AutodocDirective
         self.registry.add_documenter(cls.objtype, cls)
         self.add_directive('auto' + cls.objtype, AutodocDirective)
 
     def add_autodoc_attrgetter(self, typ, getter):
         # type: (Type, Callable[[Any, unicode, Any], Any]) -> None
+        """Register a new ``getattr``-like function for the autodoc extension.
+
+        Add *getter*, which must be a function with an interface compatible to
+        the :func:`getattr` builtin, as the autodoc attribute getter for
+        objects that are instances of *type*.  All cases where autodoc needs to
```

Comment:
Now we use `typ` instead. Could you update this please?

Reply:
Done